internal/migrate/get_staged_code.go

```diff
@@ -64,24 +65,56 @@ func getStagedCode(
 		return nil, fmt.Errorf("error getting address by contract name: %w", err)
 	}
 
-	cName, err := cadence.NewString(contractName)
+	location := common.NewAddressLocation(nil, common.Address(addr), contractName)
+	code, err := getStagedContractCode(context.Background(), flow, location)
 	if err != nil {
-		return nil, fmt.Errorf("error creating cadence string: %w", err)
+		return nil, err
+	}
+
+	// If the contract is not staged, return nil
+	if code == nil {
+		return scripts.NewScriptResult(cadence.NewOptional(nil)), nil
 	}
 
-	caddr := cadence.NewAddress(addr)
+	return scripts.NewScriptResult(cadence.NewOptional(cadence.String(code))), nil
```

Comment:
Note: this has changed because I tried to extract some of this common functionality.  The result is converted back into Cadence values just so that we can retain the same output (e.g. `scripts.NewScriptResult`).  It's probably not optimal, and probably makes more sense to have a custom output for this command IMO, but out of scope for this PR.